operator/docs/enhancements/zone_aware_replication.md

```diff
@@ -0,0 +1,460 @@
+  annotations: 
+    topology.kubernetes.io/zone: zone-a 
+spec:
+  topologySpreadConstraints:
+  - maxSkew: 1
+    topologyKey: topology.kubernetes.io/zone
+    whenUnsatisfiable: DoNotSchedule
+    labelSelector:
+      matchLabels:
+        ingester: pod
+  containers:
+  - name: ingester
+    env:
+    - name: ZONE
+      value: ""
+    command: ["sh", "-c"]
+        args:
+        - while true; do
+            if [[ -e /etc/podinfo/annotations ]]; then
+              if [[ -s /etc/podinfo/annotations ]]; then
+                echo -en '\n\n'; cat /etc/podinfo/annotations;
+                ZONE=$(cat /etc/podinfo/annotations); 
+              else
+                echo "Empty File"; fi; fi;
+            sleep 5;
+          done;
+    volumeMounts:
+      - name: podinfo
+        mountPath: /etc/podinfo
+  volumes:
+    - name: podinfo
+      downwardAPI:
+        items:
+          - path: "annotations"
+            fieldRef:
+              fieldPath: metadata.annotations['topology.kubernetes.io/zone']
+```
+
+##### Drawbacks of Solution a
+
+- The status of the application pod during the waiting phase is unknown
+- The liveness&readiness probes will fail since the loki application would not have started
+
+#### b. Introduce an Init Container
+
+In this approach we introduce a conditional init container which is used only if `zone-aware` is enabled. The Init container has the task of checking for the pod annotation to be set to the topologykey. Once this value is succesfully set, the main application container is started, and an ENV variable is set that is used in the loki-config.yaml. The loki application is then started. In this way we can be sure that the loki application will have the domain information
+
+```yaml
+kind: Pod
+apiVersion: v1
+metadata:
+  name: ingester
+  annotations: 
+    topology.kubernetes.io/zone: zone-a 
+spec:
+  topologySpreadConstraints:
+  - maxSkew: 1
+    topologyKey: topology.kubernetes.io/zone
+    whenUnsatisfiable: DoNotSchedule
+    labelSelector:
+      matchLabels:
+        ingester: pod
+  initContainers:
+  - name: init-envval
+    image: busybox:1.28
+    command: ['sh', '-c', "until cat /etc/podinfo/annotations; do echo waiting for toppologykey; sleep 2; done"]
+    volumeMounts:
+      - name: podinfo
+        mountPath: /etc/podinfo
+  containers:
+  - name: ingester
+    env:
+    - name: ZONE
+      valueFrom:
+        fieldRef:
+          fieldPath: metadata.annotations['topology.kubernetes.io/zone']
+  volumes:
+    - name: podinfo
+      downwardAPI:
+        items:
+          - path: "annotations"
+            fieldRef:
+              fieldPath: metadata.annotations['topology.kubernetes.io/zone']
+```
+
+##### Drawbacks of Solution b
+
+- Maintain an image for the  new Init Container. - Since the Init container's work is very basic. We could try to see if we can use an existing lightweight container image for our purpose. That way we can avoid the additional work of maintaining a new image.
```

Comment:
```suggestion
- Maintain an image for the new init-container. Since the init-container's work is very basic, ee could try to see if we can use an existing lightweight container image for our purpose. That way we can avoid the additional work of maintaining a new image.
```